Solutions/Marain.Workflow.Api.MessageProcessingHost/Marain/Workflows/Api/MessageProcessingHost/Activities/CompleteLongRunningOperationActivity.cs

```diff
@@ -42,12 +48,14 @@ public CompleteLongRunningOperationActivity(IMarainOperationsControl operationsC
             [ActivityTrigger] IDurableActivityContext context)
         {
             (Guid operationId, string tenantId) = context.GetInput<(Guid, string)>();
+            string delegatedTenantId =
+                await this.marainServicesTenancy.GetDelegatedTenantIdForRequestingTenantAsync(tenantId);
```

Comment:
I love how all the hidden complexity boiled down to to this lovely and quite clear line of code for the developer to say "give me *my* delegated service tenant for the my tenant making this request". Traceability. Partitioning. Joy.